js/lib/colormaps.js

```diff
@@ -0,0 +1,58 @@
+var widgets = require('@jupyter-widgets/base');
+var ipydatawidgets = require('jupyter-dataserializers');
+var yt_tools = require('yt-tools');
+
+var CMapModel = widgets.WidgetModel.extend({
+
+    defaults: function() {
+        return _.extend(widgets.WidgetModel.prototype.defaults.call(this), {
+            _model_name: 'CMapModel',
+            _model_module: 'yt-jscanvas',
+            _model_module_version: '0.1.0',
+
+            cmaps: undefined,
+        });
+    },
+
+    initialize: function() {
+        console.log('initializing colormaps object in WASM');
+    },
+
+    normalize: function(name, buffer, take_log) {
+        // normalizes a given buffer with a colormap name. Requires colormaps
+        // to be loaded in to wasm, so requires add_mpl_colormaps to be called 
+        // at this time. 
+        return this.add_mpl_colormaps_to_wasm().then(function(colormaps) {
+            array = colormaps.normalize(name, buffer, take_log);
+            return array
+        });
+    },
+
+    add_mpl_colormaps_to_wasm: function() {
+        // initializes the wasm colormaps module from yt tools and adds the 
+        // arrays stored in the self.cmaps dict on the python side into
+        // the colormaps object in wasm.
+        
+        return yt_tools.booted.then(function(yt_tools) {
```

Comment:
Without changing a *lot*, I think you could make this early terminate if they've already been added.  In that case, just check if `this.colormaps` has been finalized or whatever, and return it.  But another option would be to move most of this into `initialize`, and inside that, set `this.colormaps` equal to a promise by doing `this.colormaps = yt_tools.booted.then( ... )`.  Then you could always just access it as `this.colormaps.then()` wherever you need it.